### PR TITLE
[CSM] Add function to get player names in range

### PR DESCRIPTION
**Pull request by bigfoot547**
This function called `get_names_in_range()` returns a list of strings of player names sent to the client.

### CODE REVIEW

**Comment by bigfoot547** [comment]
@nerzhul Please review :smile: .

File: src/script/lua_api/l_client.cpp [hunk 1]
```diff
@@ -69,6 +69,23 @@ int ModApiClient::l_display_chat_message(lua_State *L)
 	return 1;
 }
 
+// get_names_in_range()
+int ModApiClient::l_get_names_in_range(lua_State *L)
+{
+	const std::list<std::string> plist = getClient(L)->getConnectedPlayerNames();
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: const ref

File: src/script/lua_api/l_client.cpp [hunk 2]
```diff
@@ -69,6 +69,23 @@ int ModApiClient::l_display_chat_message(lua_State *L)
...
+	lua_createtable(L, plist.size(), 0);
+	int newTable = lua_gettop(L);
+	int index = 1;
+	std::list<std::string>::const_iterator iter = plist.begin();
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: please use a for loop

File: doc/client_lua_api.md [hunk 3]
```diff
@@ -737,6 +737,10 @@ Call these functions only at load time!
 * `minetest.get_wielded_item()`
     * Returns the itemstack the local player is holding
 
+### Client Environment
+* `minetest.get_names_in_range()`
```
Comment (nerzhul) [inline doc/client_lua_api.md]: get_player_names_in_range() is better

**Comment by red-001** [comment]
Maybe you could change the name? `get_names_in_range()` seems to suggest that you can pass a range to the function.

**Comment by bigfoot547** [comment]
@nerzhul I updated the branch.
@red-001 Maybe `get_player_names()`?

**Comment by SmallJoker** [comment]
`minetest.get_player_names` sounds good, take that one.

**Comment by bigfoot547** [comment]
I changed the name! Ready for merge.

File: doc/client_lua_api.md [hunk 4]
```diff
@@ -738,7 +738,7 @@ Call these functions only at load time!
     * Returns the itemstack the local player is holding
 
 ### Client Environment
-* `minetest.get_player_names_in_range()`
+* `minetest.get_player_names()`
     * Returns list of player names on server in range
```
Comment (SmallJoker) [inline doc/client_lua_api.md]: `Returns the list of all player names on the server`

**Comment by bigfoot547** [comment]
@red-001 Added.
@SmallJoker Changed.

**Comment by nerzhul** [comment]
okay for me

**Pull request merged**